File: app/src/main/java/io/github/husseinfo/countin/MainActivity.java
package io.github.husseinfo.countin;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;

import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

public class MainActivity extends Activity {
    private RecyclerView countsRecyclerView;
    private CountsListAdapter countsListAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.fab_add).setOnClickListener(v -> startActivity(new Intent(this, AddItemActivity.class)));

        this.countsRecyclerView = findViewById(R.id.counts);
        this.countsListAdapter = new CountsListAdapter();
        this.countsRecyclerView.setAdapter(this.countsListAdapter);
        this.countsRecyclerView.setLayoutManager(new LinearLayoutManager(this, LinearLayoutManager.HORIZONTAL, false));
        this.countsRecyclerView.setHasFixedSize(false);
    }

    @Override
    protected void onResume() {
        super.onResume();
        this.countsListAdapter.update(Storage.getItems(this));
    }
}

File: app/src/main/java/io/github/husseinfo/countin/MainActivity.kt
package io.github.husseinfo.countin

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import io.github.husseinfo.countin.data.AppDatabase
import kotlinx.coroutines.*

class MainActivity : Activity(), CoroutineScope by MainScope() {
    private lateinit var countsRecyclerView: RecyclerView
    private lateinit var countsListAdapter: CountsListAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        findViewById<View>(R.id.fab_add).setOnClickListener { v: View? ->
            startActivity(
                Intent(
                    this,
                    AddItemActivity::class.java
                )
            )
        }
        countsRecyclerView = findViewById(R.id.counts)
        countsListAdapter = CountsListAdapter()
        countsRecyclerView.adapter = countsListAdapter
        countsRecyclerView.layoutManager = LinearLayoutManager(
            this,
            LinearLayoutManager.VERTICAL,
            false
        )
        countsRecyclerView.setHasFixedSize(false)
    }

    override fun onResume() {
        super.onResume()

        val items = async(Dispatchers.IO) {
            AppDatabase.getDb(baseContext)!!.countDAO()?.all!!
        }

        launch(Dispatchers.Main) {
            countsListAdapter.update(items.await())
        }
    }
}
